packages/actions/test/e2e/01-setup.test.ts

```diff
@@ -1,58 +1,225 @@
+    // storing the uid so we can delete the user after the test
+    let coordinatorUid: string
+
+    // generate passwords for user and coordinator
+    const userPwd = generatePseudoRandomStringOfNumbers(24)
+    const coordinatorPwd = generatePseudoRandomStringOfNumbers(24)
+
+    // Initialize user and admin services.
+    const { userApp, userFunctions } = initializeUserServices()
+    const { adminFirestore, adminAuth } = initializeAdminServices()
+    const userAuth = getAuth(userApp)
+
+    // Get configs for storage.
+    const { ceremonyBucketPostfix } = getStorageConfiguration()
+    const ceremonyData = fakeCeremoniesData.fakeCeremonyScheduledFixed
+    const bucketName = getBucketName(ceremonyData.data.prefix, ceremonyBucketPostfix)
+    // file to upload
+    const localPath = "/tmp/circuitMetadata.log"
+    fs.writeFileSync(localPath, "test data")
+    const objectName = "zkey.zkey"
+    const duplicateBucketName = randomBytes(10).toString("hex")
+
+    beforeAll(async () => {
+        // create a new user without contributor privileges
+        await createNewFirebaseUserWithEmailAndPw(userApp, user.data.email, userPwd)
+        await sleep(5000)
+
+        // Retrieve the current auth user in Firebase.
+        const currentAuthenticatedUser = getCurrentFirebaseAuthUser(userApp)
+        user.uid = currentAuthenticatedUser.uid
+
+        // create account for coordinator
+        await createNewFirebaseUserWithEmailAndPw(userApp, coordinatorEmail, coordinatorPwd)
+        await sleep(5000)
+
+        const currentAuthenticatedCoordinator = getCurrentFirebaseAuthUser(userApp)
+        coordinatorUid = currentAuthenticatedCoordinator.uid
+
+        // add custom claims for coordinator
+        await setCustomClaims(adminAuth, coordinatorUid, { coordinator: true })
+    })
+
+    it("should fail to create a sample ceremony without being a coordinator", async () => {
+        await signInWithEmailAndPassword(userAuth, user.data.email, userPwd)
+        assert.isRejected(createS3Bucket(userFunctions, bucketName))
+    })
+
+    // run these tests only in production mode
+    if (envType === TestingEnvironment.PRODUCTION) {
+        it("should revert when trying to create a ceremony with an existing prefix", async () => {
+            // login with coordinator creds
+            await signInWithEmailAndPassword(userAuth, coordinatorEmail, coordinatorPwd)
+            const currentAuthenticatedCoordinator = getCurrentFirebaseAuthUser(userApp)
+            // refresh token
+            await currentAuthenticatedCoordinator.getIdToken(true)
+            // Create once
+            await createS3Bucket(userFunctions, duplicateBucketName)
+            await sleep(5000)
+            // Create again
+            expect(createS3Bucket(userFunctions, duplicateBucketName)).to.be.rejectedWith("Failed request.")
+        })
+
+        it("should do a full multi part upload", async () => {
```

Comment:
It would seem more like a unit test than e2e. What would be the test scenario?

Reply:
Fair point, I'll move it over.